transaction/transaction.go

```diff
@@ -358,9 +366,187 @@ func MakeAssetConfigTxn(account string, feePerByte, firstRound, lastRound uint64
+
+	// Update fee
+	eSize, err := estimateSize(tx)
+	if err != nil {
+		return types.Transaction{}, err
+	}
+	tx.Fee = types.MicroAlgos(eSize * feePerByte)
+
+	if tx.Fee < MinTxnFee {
+		tx.Fee = MinTxnFee
+	}
+
 	return tx, nil
 }
 
+// MakeAssetAcceptanceTransaction creates a tx for marking an account as willing to accept the given asset
+// - account is a checksummed, human-readable address that will send the transaction and begin accepting the asset
+// - feePerByte is a fee per byte
+// - firstRound is the first round this txn is valid (txn semantics unrelated to asset management)
+// - lastRound is the last round this txn is valid
+// - note is an arbitrary byte array
+// - genesis id corresponds to the id of the network
+// - genesis hash corresponds to the base64-encoded hash of the genesis of the network
+// - creator is the address of the asset creator
+// - index is the asset index
+func MakeAssetAcceptanceTxn(account string, feePerByte, firstRound, lastRound uint64, note []byte,
+	genesisID, genesisHash, creator string, index uint64) (types.Transaction, error) {
+	tx, err := MakeAssetTransferTxn(account, account, "", 0,
+		feePerByte, firstRound, lastRound, note, genesisID, genesisHash, creator, index)
+	return tx, err
+}
+
+// MakeAssetRevocationTransaction creates a tx for revoking an asset from an account and sending it to another
+// - account is a checksummed, human-readable address; it must be the revocation manager / clawback address from the asset's parameters
+// - target is a checksummed, human-readable address; it is the account whose assets will be revoked
+// - recipient is a checksummed, human-readable address; it will receive the revoked assets
+// - feePerByte is a fee per byte
+// - firstRound is the first round this txn is valid (txn semantics unrelated to asset management)
+// - lastRound is the last round this txn is valid
+// - note is an arbitrary byte array
+// - genesis id corresponds to the id of the network
+// - genesis hash corresponds to the base64-encoded hash of the genesis of the network
+// - creator is the address of the asset creator
+// - index is the asset index
+func MakeAssetRevocationTxn(account, target, recipient string, amount, feePerByte, firstRound, lastRound uint64, note []byte,
+	genesisID, genesisHash, creator string, index uint64) (types.Transaction, error) {
+	tx, err := MakeAssetTransferTxn(account, recipient, "", amount,
+		feePerByte, firstRound, lastRound, note, genesisID, genesisHash, creator, index)
+
+	targetAddr, err := types.DecodeAddress(target)
+	if err != nil {
+		return tx, err
+	}
+	tx.AssetSender = targetAddr
+
+	// Update fee
+	eSize, err := estimateSize(tx)
+	if err != nil {
+		return types.Transaction{}, err
+	}
+
+	tx.Fee = types.MicroAlgos(eSize * feePerByte)
+	if tx.Fee < MinTxnFee {
+		tx.Fee = MinTxnFee
+	}
+
+	return tx, err
+}
+
+// MakeAssetMintTransaction creates a tx for minting more of an asset (transferring from the reserve account)
```

Comment:
Same for this one. I believe it would be much easier to simply explain this in the documentation but it worth a discussion.